src/cmd-buildextend-installer

```diff
@@ -150,17 +149,19 @@ def generate_iso():
     initramfs_img = 'initramfs.img'
 
     tmpisofile = os.path.join(tmpdir, iso_name)
-    img_metal_obj = buildmeta['images'].get('metal')
-    if img_metal_obj is None:
-        raise Exception("ISO generation requires `metal` image")
-
-    img_metal = os.path.join(builddir, img_metal_obj['path'])
-    img_metal_checksum = img_metal_obj['sha256']
 
+    img_metal_obj = buildmeta['images'].get('metal')
     img_metal4k_obj = buildmeta['images'].get('metal4k')
-    if img_metal4k_obj is not None:
+    if is_live:
+        if img_metal_obj is None or img_metal4k_obj is None:
+            raise Exception("Live image generation requires `metal` and `metal4k` images")
         img_metal4k = os.path.join(builddir, img_metal4k_obj['path'])
         img_metal4k_checksum = img_metal4k_obj['sha256']
+    else:
+        if img_metal_obj is None:
+            raise Exception("ISO image generation requires `metal` image")
```

Comment:
If we're building a legacy installer (that doesn't include the rootfs) do we really require the `metal` image?

Reply:
ahh, seems like we get the kernel arguments and such from the `img_metal` later on so I guess it is required.